### PR TITLE
Add support for custom checkpoints in MusicGen

### LINKED ISSUE #29873: Uploading own MusicGen checkpoint
### Feature request

Hi, I'm interested in the MusicGen model. I saw that you provide a conversion option for pre-trained models. But how can I load my own model? I don't see the possibility of using my own checkpoint. I carried out fine tuning from the original small model. I would be very happy to be given such an opportunity.

### Motivation

When I want to load my checkpoint using MusicgenForConditionalGeneration.from_pretrained(), I can't specify my path to it.

### Your contribution

I'm just wondering if there is a way or recipe to use my custom checkpoint.
src/transformers/models/musicgen/convert_musicgen_transformers.py

### PR DESCRIPTION
**Pull request by jla524**
# What does this PR do?

Fixes #29873 (issue)

Add a new argument for custom checkpoint and pass it into`get_pretrained`

## Who can review?

@ElizavetaSedova and @ArthurZucker

### CODE REVIEW

File: src/transformers/models/musicgen/convert_musicgen_transformers.py [hunk 1]
```diff
@@ -212,6 +212,7 @@ def convert_musicgen_checkpoint(
         "`['facebook/musicgen-stereo-small', 'facebook/musicgen-stereo-medium', 'facebook/musicgen-stereo-large']` "
         "for the stereo checkpoints.",
     )
```
Comment (ArthurZucker) [inline src/transformers/models/musicgen/convert_musicgen_transformers.py:214]: IMO we just have to update the doc here no?

Comment (ArthurZucker) [reply src/transformers/models/musicgen/convert_musicgen_transformers.py:214]: To mention that it can also be a path to a custom checkpoint

Comment (jla524) [reply src/transformers/models/musicgen/convert_musicgen_transformers.py:214]: I think `decoder_config_from_checkpoint` will throw an error with a custom checkpoint

https://github.com/huggingface/transformers/blob/3cc28d141f895f071960a0d5dfce9554ba3739a3/src/transformers/models/musicgen/convert_musicgen_transformers.py#L105-L109

Is it possible to infer the config?

Comment (ArthurZucker) [reply src/transformers/models/musicgen/convert_musicgen_transformers.py:214]: We can update that error as well, the conversion script is not meant to only support official checkpoints.

Comment (jla524) [reply src/transformers/models/musicgen/convert_musicgen_transformers.py:214]: That makes sense. The script should be flexible enough to support all checkpoints. Can we expect the custom checkpoints to follow the same naming scheme? i.e. `musicgen-stereo-[small|medium|large]`

**Comment by ArthurZucker** [comment]
Hey! Can you make sure the CI is green and ping me again for a reveiw!

**Comment by jla524** [comment]
Hey @ArthurZucker! The CI is green now.

**Review by ArthurZucker — APPROVED** [review]
Delay but thanks for the PR! 🔥

**Pull request merged**